### ISSUE TITLE
Teradata does not support the != operator - should be <>

### ISSUE THREAD

**Issue by mcloney-ddm** [sqlfluff/sqlfluff#7290]
### Search before asking

- [x] I searched the [issues](https://github.com/sqlfluff/sqlfluff/issues) and found no similar issues.

### What Happened

Used SQLFluff against a Teradata SQL file with `--sqlfluff:dialect:teradata` as the first line of the file.

SQLFluff reformatted the file and replaced a `<>` operator with `!=` per rule CV01.

### Expected Behaviour

Exactly the same behavior but preserving `<>` instead of replacing it with `!=`.

### Observed Behaviour

Teradata does not support the !=` operator, so the query did not parse in Teradata.

### How to reproduce

Run `sqlfluff fix` on a file like this:

```sql
--sqlfluff:dialect:teradata

SELECT * FROM my.table WHERE id <> 1;
```

### Dialect

Teradata

### Version

3.5.0

### Configuration

[sqlfluff]
dialect = snowflake
templater = jinja
exclude_rules = ambiguous.column_count
max_line_length = 120
large_file_skip_byte_limit = 25000
sql_file_exts = .sql,.sql.j2

[sqlfluff:rules:convention.not_equal]
preferred_not_equal_style = c_style

[sqlfluff:templater:jinja]
library_path = sqlfluff_libs

[sqlfluff:templater:jinja:context]
ds = "2025-11-10"

[sqlfluff:rules:capitalisation.keywords]
capitalisation_policy = upper

[sqlfluff:rules:capitalisation.functions]
extended_capitalisation_policy = upper

### Are you willing to work on and submit a PR to address the issue?

- [x] Yes I am willing to submit a PR!

### Code of Conduct

- [x] I agree to follow this project's [Code of Conduct](https://github.com/sqlfluff/sqlfluff/blob/main/CODE_OF_CONDUCT.md)
